metriken-exposition/src/snapshotter.rs

```diff
@@ -0,0 +1,102 @@
+use std::any::Any;
+
+use crate::*;
+
+/// Produces a snapshot of metric readings.
+pub struct Snapshotter {
+    kind_filter: fn(Option<&dyn Any>) -> bool,
+    name_filter: fn(&str) -> bool,
```

Comment:
As a suggestion: just one filter that takes a `&MetricEntry`. It can do all the operations of either of the filter functions here.

Reply:
Done.